frontend_lib/src/component/Timeline/Timeline.jsx

```diff
@@ -156,6 +228,29 @@ export class Timeline extends React.Component {
           />
         )}
 
+        {state.showEditCommentPopup && (
```

Comment:
I'm unsure whether all the 3 boolean states can be active separately. If they are mutually exclusive, a state value with named properties would make it explicit (note: you can leave it as it is for now).